client/app/components/CreateSourceDialog.jsx

```diff
@@ -0,0 +1,179 @@
+      includes(type.name.toLowerCase(), searchText.toLowerCase()));
+    return (
+      <div className="m-t-20">
+        <Search
+          placeholder="Search..."
+          onChange={e => this.setState({ searchText: e.target.value })}
+          autoFocus
+        />
+        <div className="scrollbox p-5 m-t-10" style={{ minHeight: '30vh', maxHeight: '40vh' }}>
+          {isEmpty(filteredTypes) ? (<EmptyState className="" />) : (
+            <List
+              size="small"
+              dataSource={filteredTypes}
+              renderItem={item => this.renderItem(item)}
+            />
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  renderForm() {
+    const { types, imageFolder, helpLinks } = this.props;
+    const { selectedType } = this.state;
+    const type = find(types, { type: selectedType });
+    const fields = helper.getFields(type.configuration_schema);
+    return (
+      <div className="scrollbox p-5 m-t-10" style={{ maxHeight: '50vh' }}>
+        <div className="text-center">
+          <img src={`${imageFolder}/${selectedType}.png`} alt={type.name} width="48" />
+          <h4 className="di-block">{type.name}</h4>
+          {helpLinks[selectedType] && (
+          <p className="text-center">
+            {/* eslint-disable-next-line react/jsx-no-target-blank */}
+            <a href={helpLinks[selectedType]} target="_blank" rel="noopener">
+              Help setting up {type.name} <i className="fa fa-external-link" aria-hidden="true" />
+            </a>
+          </p>
+          )}
+        </div>
+        <DynamicForm
+          id="sourceForm"
+          fields={fields}
+          onSubmit={this.createSource}
+          feedbackIcons
+          hideSubmitButton
+        />
+      </div>
+    );
+  }
+
+  renderItem(item) {
+    const { imageFolder } = this.props;
+    return (
+      <List.Item
+        className="p-l-10 p-r-10 clickable"
+        onClick={() => this.selectType(item.type)}
+      >
+        <PreviewCard title={item.name} imageUrl={`${imageFolder}/${item.type}.png`}>
+          <i className="fa fa-angle-double-right" />
+        </PreviewCard>
+      </List.Item>
+    );
+  }
+
+  render() {
+    const { currentStep, savingSource } = this.state;
+    const { dialog, sourceType } = this.props;
+    return (
+      <Modal
+        {...dialog.props}
+        title={`Create a New ${sourceType}`}
+        footer={(currentStep === StepEnum.SELECT_TYPE) ? [
+          (<Button key="cancel" onClick={() => dialog.dismiss()}>Cancel</Button>),
+          (<Button key="submit" type="primary" disabled>Create</Button>),
+        ] : [
+          (<Button key="previous" onClick={this.resetType}>Previous</Button>),
+          (
+            <Button key="submit" htmlType="submit" form="sourceForm" type="primary" loading={savingSource}>
+              Create
+            </Button>
+          ),
+        ]}
+      >
+        <Steps className="hidden-xs" size="small" current={currentStep} progressDot>
```

Comment:
I had to hide this for mobiles because it was overflowing the dialog, I've opened ant-design/ant-design#15215 for this. I should add a comment to track this in case we want this removed when fixed, but perhaps this is better hidden for mobiles as it increases height that we already don't have much.